opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/auto_instrumentation.py

```diff
@@ -15,22 +15,33 @@
 # limitations under the License.
 
 from logging import getLogger
-from runpy import run_path
+from os import environ, execl
+from os.path import abspath, dirname, pathsep
+from shutil import which
 from sys import argv
 
-from pkg_resources import iter_entry_points
-
 logger = getLogger(__file__)
 
 
 def run() -> None:
 
-    for entry_point in iter_entry_points("opentelemetry_instrumentor"):
-        try:
-            entry_point.load()().instrument()  # type: ignore
-            logger.debug("Instrumented %s", entry_point.name)
+    python_path = environ.get("PYTHONPATH", [])
+
+    if not python_path:
+        python_path = []
+
+    if python_path:
+        python_path = python_path.split(pathsep)
+
+    filedir_path = dirname(abspath(__file__))
+
+    if filedir_path in python_path:
+        python_path.remove(filedir_path)
+
+    python_path.insert(0, filedir_path)
+
+    environ["PYTHONPATH"] = pathsep.join(python_path)
```

Comment:
Do we really need to guarantee that the element is unique in PYTHONPATH?
I think the following could have the same result.

```
orig_path = environ.get("PYTHONPATH", "")
os.environ["PYTHONPATH"] = filedir_path + pathsep.join + orig_path
```

Reply:
I prefer so, we are adding a `sitecustomize.py` file in the `PYTHONPATH`, I imagine very weird bugs may happen if this is file is loaded more than once.